### ISSUE TITLE
[Mobile Payments] New order notice can appear after accepting payment for an order

### ISSUE THREAD

**Issue by rachelmcr** [woocommerce/woocommerce-ios#4582]
**Describe the bug**

I noticed that after I accepted a payment for an order, I saw the "You have a new order!" notice in the app. This wouldn't be a big deal except that this notice also has haptic feedback.

(I confirmed that I can reproduce this consistently and it isn't happening because some other order happens to come in at the same time.)

**To Reproduce**
1. Make sure your store is set up to accept payments.
2. Create an order on your store with a `Pending payment` status.
3. Open the order detail for that order in the app.
4. Tap the "Collect payment" button.
5. Set up your card reader if needed and accept a payment for the order.
6. Notice that underneath the modal, the new order notice appears and you get haptic feedback.

**Screenshots**

<img src="https://user-images.githubusercontent.com/8658164/124968082-1c653400-e01d-11eb-925e-df6aed2b09d7.PNG" width="300px">

**Comment by ctarda** [comment]
I haven't been able to reproduce this. Is this still happening @rachelmcr ?

**Comment by rachelmcr** [comment]
Yep, I can still reproduce this! However, I did a bit more testing and realized that it isn't specific to payments. It looks like maybe orders with `Pending Payment` status don't trigger a push notification (at least, they don't seem to in my testing). When the order changes status to `Completed` (whether it's a manual status change or automatic, e.g. when accepting a payment) that triggers a new push notification.

Here's a screencast in case it helps:

https://user-images.githubusercontent.com/8658164/130454858-f8bdde10-cd65-445e-aba0-6ec2fb902263.MP4

**Comment by ctarda** [comment]
Thanks for looking into this @rachelmcr . I still can not reproduce this, but will continue testing, thanks!

**Comment by ctarda** [comment]
This is not directly related to In-Person Payments, so I will be moving it to Continuous Improvements

**Comment by thehenrybyrd** [comment]
I ran into this in testing for Order Creation as well, though I wasn't able to reproduce it as consistently.

**Comment by jaclync** [comment]
I wasn't able to reproduce this after collecting payment for a few orders (status "Pending payment") with Order Creation

**Comment by rachelmcr** [comment]
I checked and I can still reproduce this (in the TestFlight 9.1.0.0 beta release) after creating a "Pending payment" order with a product using in-app Order Creation, and then collecting payment.

I tried several times with different products and see the notice consistently. I also confirmed that when I tap "View" on the notice it reopens the same order that I just collected payment for.

<img src="https://user-images.githubusercontent.com/8658164/167112577-1e97371d-3231-41c6-a2d9-e619dce0ac0d.PNG" width="300px">

**Comment by jaclync** [comment]
Thanks for the helpful info @rachelmcr! I tested again with the 9.1 TestFlight build, then the latest debug build from `trunk`, and I was able to reproduce consistently now (maybe the push notifications weren't working for some reason when I tested it last week). I agree with your comment https://github.com/woocommerce/woocommerce-ios/issues/4582#issuecomment-903770637 that the push notification comes from the order status change from "Pending payment" to "Completed," and the logic is on the push notifications backend.

What do you think is the expected behavior? I think it's fine to have an in-app notice after payment collection with a cha-ching sound, but the copy "You have a new order!" is confusing. Maybe we can:
- Don't send a push notification if the order status change is triggered by IPP (this is probably easier to achieve on the backend since we don't have the order ID from the push notification payload right away)
- Show a different copy for the notice title along the line of "Your order is paid" if the order status change is triggered by IPP (again, this is easier to implement on the backend)

I tried thinking if we can implement something without backend changes, but haven't thought of any good ideas 🤔

**Comment by rachelmcr** [comment]
> I think it's fine to have an in-app notice after payment collection with a cha-ching sound, but the copy "You have a new order!" is confusing.

Agreed! I think either of your suggestions would be an improvement over the current behavior. Since the IPP modal already has a "Payment successful" message I don't think we really need a notice at all in this case, so I'd lean toward not sending a push notification at all in this case.

**Comment by jaclync** [comment]
Not sending a push notification sounds like a good solution 👍 this involves backend changes, I added the closest label `needs: api support` for this issue to be on Flamingo's radar even though it's not an API request

**Comment by joshheald** [comment]
Automated triage note: this was tested and commented by Codex using device/simulator automation.

I could not complete this payment-accepted repro on an iPhone 13 Pro running iOS 18.7.1 with Woo (Dev) 24.9 (24.9.0.4). I selected Card Reader and the app discovered a reader, but connection failed with a Bluetooth disconnect and retry stayed on Scanning for reader. Since I could not accept a card-reader payment, I could not verify whether a new order notice still appears afterward.

Leaving open as not fully tested.

**Comment by dangermattic** [comment]
Thanks for reporting! 👍

**Comment by iamgabrielma** [comment]
Recreated in Linear as WOOMOB-3683. Closing here as part of migrating mobile issue tracking from GitHub to Linear.

_This action was performed by an automated triage process run by @iamgabrielma. If this was closed in error, please reopen._
